src/features/blogs/blogsSlice.js

```diff
@@ -0,0 +1,42 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+
+const INITIAL_STATE = {
+  blogs: JSON.parse(localStorage.getItem("blogs")) || [],
+  isLoading: false,
+  isError: false,
+  error: "",
+};
+
+export const postBlog = createAsyncThunk("blog/postBlog", async (blog) => {
+  const existingBlogs = JSON.parse(localStorage.getItem("blogs")) || [];
```

Comment:
```suggestion
  const existingBlogs = await JSON.parse(localStorage.getItem("blogs")) || [];
```
no async without await please